jwst/tests_nightly/general/nirspec/test_nirspec_steps_single.py

```diff
@@ -168,3 +172,358 @@ def test_nrs_fs_single_spec2(self):
+            # for this slit subtract the background subtracted data from
+            # the science data with no background added
+            sub = result_slit - sci
+            # do not look at outliers - they confuse things
+            sub_valid = sub[valid]
+            mean_sub = np.mean(sub_valid[mask_clean])
+            atol = 0.5
+            assert_allclose(np.absolute(mean_sub), 0, atol=atol)
+        input_sci.close()
+        # ______________________________________________________________________
+        # Test 3 Compare background sutracted science data (results)
+        #  to a truth file. This data is MultiSlit data
+        result_file = result.meta.filename
+        result.save(result_file)
+        result.close()
+        ref_file = self.get_data(*self.ref_loc,
+                                  'nrs_sci+bkg_masterbackgroundstep.fits')
+
+        outputs = [(result_file, ref_file)]
+        self.compare_outputs(outputs)
+
+
+@pytest.mark.bigdata
+class TestNIRSpecMasterBackground_IFU(BaseJWSTTest):
+    input_loc = 'nirspec'
+    ref_loc = ['test_masterbackground', 'nrs-ifu', 'truth']
+    test_dir = ['test_masterbackground', 'nrs-ifu']
+
+    def test_nirspec_masterbackground_ifu_user1d(self):
+        """
+
+        Regression test of master background subtraction for NRS IFU when a user 1-D spectrum is provided.
+
+        """
+        # input file has 2-D background image added to it
+
+        input_file = self.get_data(*self.test_dir,
+                                    'prism_sci_bkg_cal.fits')
+        # user provide 1-D background was created from the 2-D background image
+        input_1dbkg_file = self.get_data(*self.test_dir,
+                                          'prism_bkg_x1d.fits')
+
+        result = MasterBackgroundStep.call(input_file,
+                                           user_background=input_1dbkg_file)
+
+        # _________________________________________________________________________
+        # Test 1 compare extracted spectra data with
+        # no background added to extracted spectra from the output
+        # from MasterBackground subtraction. First cube_build has to be run
+        # on the data.
+
+        result_s3d = CubeBuildStep.call(result)
+        # run 1-D extract on results from MasterBackground step
+        result_1d = Extract1dStep.call(result_s3d, subtract_background=False)
+
+        # run 1-D extract on original science data without background
+        input_sci_cal_file = self.get_data(*self.test_dir,
+                                            'prism_sci_cal.fits')
+        # find the 1D extraction of this file
+        # find Extract1dStep on sci data to use the same version of
+        # this rountine run on both files  rather than running it off line
+        # and having the 1-D extracted science file stored as input to step
+
+        sci_s3d = CubeBuildStep.call(input_sci_cal_file)
+        sci_1d = Extract1dStep.call(sci_s3d, subtract_background=False)
```

Comment:
Is there a reason that `CubeBuildStep` and `Extract1dStep` are called here instead of just storing the built, extracted file on Artifactory? It will really speed up the test if we don't have to do this on the fly every time.